internal/service/elasticache/cluster.go

```diff
@@ -769,25 +770,26 @@ func deleteCacheCluster(ctx context.Context, conn *elasticache.Client, cacheClus
 		input.FinalSnapshotIdentifier = aws.String(finalSnapshotID)
 	}
 
+	// TODO: Migrate to retry.Operation
 	log.Printf("[DEBUG] Deleting ElastiCache Cache Cluster: %s", cacheClusterID)
-	err := retry.RetryContext(ctx, 5*time.Minute, func() *retry.RetryError {
+	err := sdkretry.RetryContext(ctx, 5*time.Minute, func() *sdkretry.RetryError {
 		_, err := conn.DeleteCacheCluster(ctx, input)
 		if err != nil {
 			if errs.IsAErrorMessageContains[*awstypes.InvalidCacheClusterStateFault](err, "serving as primary") {
-				return retry.NonRetryableError(err)
+				return sdkretry.NonRetryableError(err)
 			}
 			if errs.IsAErrorMessageContains[*awstypes.InvalidCacheClusterStateFault](err, "only member of a replication group") {
-				return retry.NonRetryableError(err)
+				return sdkretry.NonRetryableError(err)
 			}
 			// The cluster may be just snapshotting, so we retry until it's ready for deletion
 			if errs.IsA[*awstypes.InvalidCacheClusterStateFault](err) {
-				return retry.RetryableError(err)
+				return sdkretry.RetryableError(err)
 			}
-			return retry.NonRetryableError(err)
+			return sdkretry.NonRetryableError(err)
 		}
 		return nil
 	})
-	if tfresource.TimedOut(err) {
+	if retry.TimedOut(err) {
```

Comment:
I'm thinking now it's probably best to keep this as `tfresource.TimedOut`, at least until we are sure nothing uses the SDKv2 `helper/retry` package.